references.go

```diff
@@ -159,84 +156,28 @@ func blobHash(path string, commit *object.Commit) (hash plumbing.Hash, found boo
 	return file.Hash, true
 }
 
-type contentsComparatorFn func(path string, a, b *object.Commit) (bool, error)
-
 // Returns a new slice of commits, with duplicates removed.  Expects a
-// sorted commit list.  Duplication is defined according to "comp".  It
-// will always keep the first commit of a series of duplicated commits.
-func removeComp(path string, cs []*object.Commit, comp contentsComparatorFn) ([]*object.Commit, error) {
-	result := make([]*object.Commit, 0, len(cs))
+// sorted commit list.  Duplication is defined according to the blob hash.
+// It will always keep the first commit of a series of duplicated commits.
+func removeRedundant(path string, cs []*object.Commit) ([]*object.Commit, error) {
 	if len(cs) == 0 {
-		return result, nil
+		return cs, nil
 	}
+	result := make([]*object.Commit, 0, len(cs))
 	result = append(result, cs[0])
+	var hashes []plumbing.Hash
+	for _, c := range cs {
+		h, _ := blobHash(path, c)
```

Comment:
As we are ignoring the `found` value, I would add a comment to explain why that isn't a problem.